### PR TITLE
fix qmp_command.qmp_query-cpus

### PR DESCRIPTION
**Pull request by hjydxy**
Hmp command return the PC value in hexadecimal.
Qmp command return the PC value in complement.
This patch transform the value from it's hexadecimal to it's complement.

### CODE REVIEW

**Comment by xutian** [comment]
@hjydxy  no need to close it then open a new one.  push your local branch with '--force' option , commit in your pull-requests will update.

Thanks,
Xu

**Comment by hjydxy** [comment]
OK. I am not familiar with git. Think you for your remind.

**Comment by hjydxy** [comment]
Thank you ,actual ...

**Comment by hjydxy** [comment]
Sorry , i am not familiar with git . Thank you for your remind.

**Comment by hjydxy** [comment]
@xutian Please check this patch and give me some advise,thank you!

**Comment by hjydxy** [comment]
ping..

File: qemu/tests/qmp_command.py [hunk 1]
```diff
@@ -114,8 +114,21 @@ def check_result(qmp_o, output=None, exception_list=""):
                             continue
                         if '0x' in val:
                             val = long(val, 16)
-                            if val != qmp_o[i][key]:
-                                msg += "\nValue in human monitor: '%s'" % val
+                            binvalue = bin(val)
+                            binstr = str(binvalue)
+                            l = len(binstr)
```
Comment (balamuruhans) [inline qemu/tests/qmp_command.py]: please use a proper variable name here and elsewhere

**Comment by hjydxy** [comment]
@balamuruhans Is that batter?

File: qemu/tests/qmp_command.py [hunk 2]
```diff
@@ -167,50 +177,50 @@ def check_result(qmp_o, output=None, exception_list=""):
...
-                                         "or higher than no_cpus).\n%s"
-                                         % (cpu, out))
+                    raise exceptions.TestFail("Incorrect CPU index '%s' (corrupted "
+                                              "or higher than no_cpus).\n%s"
+                                              % (cpu, out))
 
     qemu_binary = utils_misc.get_qemu_binary(params)
     if not utils_misc.qemu_has_option("qmp", qemu_binary):
-        raise error.TestNAError("Host qemu does not support qmp.")
+        raise exceptions.TestNAError("Host qemu does not support qmp.")
```
Comment (balamuruhans) [inline qemu/tests/qmp_command.py]: it is `exceptions.TestSkipError`

Comment (hjydxy) [inline qemu/tests/qmp_command.py]: @balamuruhans Modified

**Comment by balamuruhans** [comment]
Looks good to me

**Comment by hjydxy** [comment]
@balamuruhans Could you please to merge this patch if it's ok.

**Pull request merged**